vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt

```diff
@@ -214,14 +214,10 @@ class UserListViewModel @AssistedInject constructor(
                 ThreePidUser(email = search, user = null)
             } else {
                 try {
-                    val user = session.getProfileAsUser(foundThreePid.matrixId)
+                    val user = tryOrNull { session.getProfileAsUser(search) } ?: User(search)
```

Comment:
I didn't look at the whole code, `search` is for an email, you should pass `foundThreePid.matrixId`, as before